packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -28,6 +28,57 @@ const int _kDefaultSizeBytes = 100 << 20; // 100 MiB
 ///
 /// A shared instance of this cache is retained by [PaintingBinding] and can be
 /// obtained via the [imageCache] top-level property in the [painting] library.
+///
+/// {@tool sample}
+///
+/// This sample shows how to supply your own caching logic and replace the
+/// global [imageCache] varible.
+///
+/// ```dart
+/// import 'package:flutter/foundation.dart' as foundation;
```

Comment:
using "as foo" is kinda un-idiomatic for flutter and clutters up the example. If you want to show where the things used in the example are coming from maybe import them as "import 'package:foo/foo.dart' show Bar;"?

Reply:
Done, fyi there are 245 instances of using `import as` in the flutter codebase =)